### ISSUE TITLE
Simplify and speed testing

### ISSUE THREAD

**Issue by billdenney** [nlmixr2/rxode2#359]
> If the `rx$` convention isn't needed any longer, perhaps we should remove all of these (and figure out what version of `testthat` where this changed).  We could also add a `pkgload::load_all()` to the tests so we don't have to worry about drifting `testthat` changes.

_Originally posted by @mattfidler in https://github.com/nlmixr2/rxode2/issues/356#issuecomment-1304872855_

**Comment by billdenney** [comment]
If this works for you, @mattfidler, I think that it would be better as the tests would be easier to read and edit, but I don't want to cause issues for your normal development path.

Another thing that could be considered is speeding up testing with parallelization.  Running the tests in parallel speeds it up a lot on my system.  There is at least one set of tests that appears to fail in parallel execution which works when run sequentially, but I think that those are worth fixing the fact that they are having the errors (which is either a code bug that we should find or a testing bug which would be best to improve) rather than run everything sequentially.  I'll try to resolve those.

What I'll add to the `DESCRIPTION` file for parallelization is the following:

```
Config/testthat/parallel: true
Config/testthat/start-first: lincmt*, random*
```

**Comment by mattfidler** [comment]
@billdenney 

I only used the `rx$` paradigm because the tests ran in `devtools::test()` but didn't run correctly in `devtools::check()`, no other reason.  If this is no longer needed then it does make the code more readable to drop this.  Though I would try to depend on the `testthat` where this changed (or use the `load_all()`). 

The only concern I have with this is that testing may assume functions are available during testing and when they are properly exported `rxode2` or `nlmixr2` doesn't work correctly.

I think it is generally safe to assume this will be Ok, though.

I am also unclear when this changed.

**Comment by mattfidler** [comment]
As far as parallelization, I think it is generally OK, though `lincmt` and `random` didn't make my list last time.  The tests that start with `test-000` were the ones that couldn't run in parallel and the `zzz` seemed to perform best at the end.  I would create a separate issue for each since they could break `rxode2` in different ways

**Comment by mattfidler** [comment]
Also there are some other tests that shouldn't be run in parallel, like the parallel ode solving.  The parallel processing in R would make them single threaded on most linux systems (at the very least) and will not test for memory corruption.  This includes parallel random number generation too, I guess.

**Comment by mattfidler** [comment]
My problem is that you cannot have a set of tests that are non parallelized for testthat (at least I couldn't find a way), otherwise I would have done it long ago.

**Comment by billdenney** [comment]
As I'm running the tests in parallel for the first time (in a while), I see a few observations:

* No surprise, `lincmt-parse` and `lincmt-solve` take the longest to solve.  Running with 12 cores, they take longer than all the other tests.  It may be worth splitting them up into smaller tests.
    * `lincmt-parse` looks like it could pretty easily be split into each of the tests indicated by the commented out `context()` lines and one more "other" for the extra stuff at the top.
    * `lincmt-solve` has what appears to be similar `context()` lines, but the tests will take a bit more effort to separate.
* Some tests appear to be failing due to missing directories.  My first guess is that running `rxClean()` may be breaking those.

**Comment by billdenney** [comment]
Thanks for those explanations, let's move parallel testing to another issue then.  These are separate concerns from the hopefully simpler removal of `rx$`.

**Comment by mattfidler** [comment]
`lincmt-parse` could possibly be ported to the `rxode2parse` package since that is where the parsing actually occurs and will boost its coverage.

`lincmt-solve` should probably also be kicked out with references from the ODEs saved.  But that would require a `rxode2lincmt` package.  The code for solving is in `rxode2parse` (and put there because of compilation time)
